src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift

```diff
@@ -135,34 +144,78 @@ class QRScannerCoordinator {
 	}
 
 	private func showQRCodeResult(qrCodeResult: QRCodeResult) {
-		parentViewController?.dismiss(animated: true, completion: { [weak self] in
-			switch qrCodeResult {
-			case let .coronaTest(testRegistrationInformation):
-				self?.showScannedTestResult(testRegistrationInformation)
-			case let .certificate(certificateResult):
-				self?.showScannedHealthCertificate(certificateResult)
-			case let .traceLocation(traceLocation):
-				self?.showScannedCheckin(traceLocation)
-			}
-		})
+		switch qrCodeResult {
+		case let .coronaTest(testRegistrationInformation):
+			showScannedTestResult(testRegistrationInformation)
+		case let .certificate(certificateResult):
+			showScannedHealthCertificate(certificateResult)
+		case let .traceLocation(traceLocation):
+			showScannedCheckin(traceLocation)
+		}
 	}
 
+	// swiftlint:disable cyclomatic_complexity
```

Comment:
The warning should be resolved by moving some code into own funcs.

Reply:
Absolutely! Done 👍